### ISSUE TITLE
Block Editor: URLInput: Remove autoFocus as a default prop of URLInput

### ISSUE THREAD

**Issue by aduth** [WordPress/gutenberg#22440]
Previously: https://github.com/WordPress/gutenberg/issues/18061#issuecomment-580480464

Currently, `URLInput` assigns auto-focus to its rendered input. Auto-focus is largely discouraged ([we even enforce a rule](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-autofocus.md)) and it should not be the default behavior for such a basic component, as it limits (complicates) reusability and makes unnecessary assumptions about its rendering context. It's assumed that this was the default largely based on the assumption that the input is typically rendered within a Popover, which is not something we should be assuming.

**Task:** Avoid assigning `autoFocus` in `URLInput`. It should still be supported as an optional (opt-in) prop in cases where it is appropriate. This usage and associated cautions should be sufficiently documented.

**Comment by youknowriad** [comment]
this has been done recently.
